utils_test.go

```diff
@@ -291,3 +293,82 @@ func TestIsolationLevelMapping(t *testing.T) {
 		t.Fatalf("Expected error to be %q, got %q", expectedErr, err)
 	}
 }
+
+func parseDateTimeOld(str string, loc *time.Location) (t time.Time, err error) {
+	base := "0000-00-00 00:00:00.0000000"
+	switch len(str) {
+	case 10, 19, 21, 22, 23, 24, 25, 26: // up to "YYYY-MM-DD HH:MM:SS.MMMMMM"
+		if str == base[:len(str)] {
+			return
+		}
+		t, err = time.Parse(timeFormat[:len(str)], str)
+	default:
+		err = fmt.Errorf("invalid time string: %s", str)
+		return
+	}
+
+	// Adjust location
+	if err == nil && loc != time.UTC {
+		y, mo, d := t.Date()
+		h, mi, s := t.Clock()
+		t, err = time.Date(y, mo, d, h, mi, s, t.Nanosecond(), loc), nil
+	}
+
+	return
+}
+
+func Test_parseDateTime(t *testing.T) {
```

Comment:
This should be named `TestParseDateTime`

Reply:
done